src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -732,6 +722,16 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
                     spentTransaction.SpendingDetails.CreationTime = DateTimeOffset.FromUnixTimeSeconds(block.Header.Time);
                 }
             }
+
+            if (withdrawal != null)
```

Comment:
I'm cautious about this. Why did you decide to move it?

If we're moving it here, we're also saying we expect that SpendingDetails might not be null sometimes. This seems to happen in logs, though very rarely. When this happens though, it's not going to set the new SpendingDetails correctly. There is likely a new transaction hash or so that is coming in, and if we're doing this then we should probably change this method to update the spending details if it exists.

Reply:
I have an existing wallet and that new variable is not populated in the wallet file. I hoped this will lead to more robust behavior.

Reply:
Okay, I think then that this whole method should be rebuilt, because the problem I mentioned still remains. If `SpendingDetails` is set, we should update it

Reply:
Removed these changes since you will be creating a separate PR.